pkg/sql/colexec/expr_eval.go

```diff
@@ -429,127 +429,147 @@ func EvalFilterByZonemap(
 	ctx context.Context,
 	meta objectio.ColumnMetaFetcher,
 	expr *plan.Expr,
+	zms []objectio.ZoneMap,
+	vecs []*vector.Vector,
 	columnMap map[int]int,
 	proc *process.Process,
-) (v objectio.ZoneMap) {
+) objectio.ZoneMap {
 	var err error
 	switch t := expr.Expr.(type) {
 	case *plan.Expr_C:
-		if v, err = getConstZM(ctx, expr, proc); err != nil {
-			v = objectio.NewZM(types.T_bool, 0)
+		if zms[expr.AuxId] == nil {
+			if zms[expr.AuxId], err = getConstZM(ctx, expr, proc); err != nil {
+				zms[expr.AuxId] = objectio.NewZM(types.T_bool, 0)
+			}
 		}
-		return
 	case *plan.Expr_Col:
-		v = meta.MustGetColumn(uint16(columnMap[int(t.Col.ColPos)])).ZoneMap()
-		return
+		zms[expr.AuxId] = meta.MustGetColumn(uint16(columnMap[int(t.Col.ColPos)])).ZoneMap()
 	case *plan.Expr_F:
 		var (
 			err error
 			f   *function.Function
 		)
+
+		args := t.F.Args
 		id := t.F.GetFunc().GetObj()
 		if f, err = function.GetFunctionByID(proc.Ctx, id); err != nil {
-			v = objectio.NewZM(types.T_bool, 0)
-			return
-		}
-		params := make([]objectio.ZoneMap, len(t.F.Args))
-		for i := range params {
-			params[i] = EvalFilterByZonemap(ctx, meta, t.F.Args[i], columnMap, proc)
-			if !params[i].IsInited() {
-				return params[i]
+			zms[expr.AuxId].Reset()
+			break
+		}
+
+		//if t.F.Func.ObjName == "in" {
```

Comment:
Do you want to keep this code?   Either remove it, or add enough comment on why you commented it out now.

Reply:
I'll commit my full in-filter implementation in next PR.